lib/loadObj.js

```diff
@@ -502,6 +503,34 @@ function assignDefaultMaterial(nodes, materials) {
     }
 }
 
+function assignUnnamedMaterial(nodes, materials) {
+    // If there is a material that doesn't have a name, assign that
+    // material to any primitives whose material is undefined.
+    const unnamedMaterial = materials.find(function(material) {
+        return material.name.length === 0;
+    });
+
+    if (!defined(unnamedMaterial)) {
+        return;
+    }
+
+    const nodesLength = nodes.length;
+    for (let i = 0; i < nodesLength; ++i) {
+        const meshes = nodes[i].meshes;
+        const meshesLength = meshes.length;
+        for (let j = 0; j < meshesLength; ++j) {
+            const primitives = meshes[j].primitives;
+            const primitivesLength = primitives.length;
+            for (let k = 0; k < primitivesLength; ++k) {
+                const primitive = primitives[k];
+                if (!defined(primitive.material)) {
+                    primitive.material = unnamedMaterial.name;
+                }
+            }
+        }
+    }
+}
```

Comment:
Can/should this use `ForEach.mesh` and `ForEach.meshPrimitive` from `gltf-pipeline`?

Reply:
I don't think we plan on including gltf-pipeline in this repo, so I would pass on this.